### ISSUE TITLE
Tiddlymap crashes if node filter has duplicates

### ISSUE THREAD

**Issue by flibbles** [felixhayashi/TW5-TiddlyMap#327]
**Reproduction**

Create a node filter that results in duplicates. Actually kind of tricky to do, which is why I imagine this bug has existed this long. But you can do it by creating three tiddlers:
```
title: A
field: dup
```
```
title: B
field: dup
```
```
title: dup
```
Then have a map filter like this: `[enlist[A B]get[field]]`. Tiddly map will crash immediately, and on start up.

It's throwing at line 472 in **js/lib/utils/tmap.js**. And it's throwing _deliberately_ after finding a duplicate, which is why I haven't made a pull request for this. Is there a reason why the duplicate isn't just ignored?

The workaround for this is to just stick `+[unique[]]` at the end of your node filter, which I'm doing. So no rush.

**Comment by felixhayashi** [comment]
Hi @flibbles! thanks for reporting. The method `getLookupTable` is indeed designed to throw if it cannot produce a hashmap with unique keys in order to give the developer who use this function a heads up that the property cannot be used as key (since it is not unique).

So for example 

```
utils.getLookupTable([ { id: 1, name: 'foo' }, { id: 1, name: 'bar' } ], 'id');
```

should definitely throw since it is an error that objects with the same id hold different values and being silent about this will only postpone the problem.

Nonetheless, the function can also be used without a second argument in which case it is absolutely justified to not throw since `true` is [used as value](https://github.com/felixhayashi/TW5-TiddlyMap/blob/81322038e4ecc107dc6af2e38e6cfde567fca7e9/src/plugins/felixhayashi/tiddlymap/js/lib/utils/tmap.js#L467) and it is safe to override true with true if the same key exists twice as it will not change the result.

So long story short, I would be fine with wrapping [this](https://github.com/felixhayashi/TW5-TiddlyMap/blob/81322038e4ecc107dc6af2e38e6cfde567fca7e9/src/plugins/felixhayashi/tiddlymap/js/lib/utils/tmap.js#L473) in an "if block" as such:

```
if (lookupKey) {
  // @todo use exception class
  throw new Error(`Cannot use "${idx}" as lookup table index`);
}
```

...which will effectively avoid an exception when not being used with a second argument (as in `filter.whitelist = utils.getLookupTable(filter.matches);`)

if you could do a PR that would be great, otherwise I will change code soon :)

**Comment by flibbles** [comment]
Super. I can PR. Tonight or tomorrow probably.

**Comment by felixhayashi** [comment]
Many thanks! @flibbles
